components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -459,6 +465,25 @@ private void parseAllowedScopesConfiguration(OMElement oauthConfigElem) {
         }
     }
 
+    /**
+     * Parse unsupported claims for introspection response configuration.
+     *
+     * @param oauthConfigElem oauthConfigElem.
+     */
+    private void parseUnsupportedClaimsForIntrospectionResponseConfiguration(OMElement oauthConfigElem) {
```

Comment:
Isn't it correct to configure the supported claims instead of Unsupported claims? Also what if we change the method name as parseRequiredClaimsInIntrospectionResponse because we are always supporting the claims which were mentioned in the spec. If the required claims are not specified the default claims will be returned, if AU team configure the required claims, only the configured claims will be returned. WDYT? @mefarazath  any concerns regarding this?

Reply:
We have the capability to add attributes to the introspection response through an extension (IntrospectionDataProvider). If we configure allowed claims, we have to keep updating the attributes added from the extensions. That would not be a good UX for devs